### PR TITLE
[CF-402] Stop migration at ctrl+c

### PR DESCRIPTION
**Pull request by SVilgelm**
InterruptedException is inherited from BaseException.
Fix VmRestore action in case CF did not copy the VMs info to info_backup.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3553/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:806: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
cloudferry/lib/os/compute/nova_compute.py:31: [W0611(unused-import), ] Unused 
signal_handler imported from cloudferry.lib.scheduler

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:806: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
cloudferry/lib/os/compute/nova_compute.py:31: [W0611(unused-import), ] Unused 
signal_handler imported from cloudferry.lib.scheduler

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3554/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:806: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
cloudferry/lib/os/compute/nova_compute.py:31: [W0611(unused-import), ] Unused 
signal_handler imported from cloudferry.lib.scheduler

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:805: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:805: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/cli/v1/commands.py [hunk 1]
```diff
@@ -63,4 +63,5 @@ def take_action(self, parsed_args):
             path_scenario=self.config.migrate.scenario,
             path_tasks=self.config.migrate.tasks_mapping))
         if status_error != errorcodes.NO_ERROR:
-            raise RuntimeError("Migration has been failed")
+            raise RuntimeError("Migration has been failed with error code %s" %
```
Comment (roman-verchikov) [inline cloudferry/cli/v1/commands.py]: "Migration failed with exit code {code}"

Comment (SVilgelm) [inline cloudferry/cli/v1/commands.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:805: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:807: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by antonf** [comment]
Please fix PEP8 complaints

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:800: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
needs some changes

**Comment by roman-verchikov** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
There were no python files updated. Admins, please, check pull request.

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/lib/scheduler/scheduler.py [hunk 2]
```diff
@@ -96,9 +97,9 @@ def start(self):
             with signal_handler.IgnoreInterruptHandler():
```
Comment (roman-verchikov) [inline cloudferry/lib/scheduler/scheduler.py]: I guess in that case we don't need `IgnoreInterruptHandler` anymore

Comment (SVilgelm) [inline cloudferry/lib/scheduler/scheduler.py]: Done. But i didn't get it, why did we need this?

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3880/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.scheduler.scheduler
cloudferry/lib/scheduler/scheduler.py:18: [W0611(unused-import), ] Unused import 
oslo_config.cfg
cloudferry/lib/scheduler/scheduler.py:21: [W0611(unused-import), ] Unused 
ERROR_INVALID_CONFIGURATION imported from cloudferry.lib.utils.errorcodes

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.scheduler.test_scheduler
tests/scheduler/test_scheduler.py:37: [E1136(unsubscriptable-object), 
SchedulerTestCase.test_start_scheduler] Value 's.event_start_task.call_args' is 
unsubscriptable
tests/scheduler/test_scheduler.py:39: [E1136(unsubscriptable-object), 
SchedulerTestCase.test_start_scheduler] Value 's.event_end_task.call_args' is 
unsubscriptable

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: tests/scheduler/test_scheduler.py [hunk 3]
```diff
@@ -31,7 +32,7 @@ def test_start_scheduler(self):
         s.event_start_task.return_value = True
         s.event_end_task = mock.Mock()
         s.start()
-        self.assertEqual(scheduler.NO_ERROR, s.status_error)
+        self.assertEqual(errorcodes.NO_ERROR, s.status_error)
```
Comment (roman-verchikov) [inline tests/scheduler/test_scheduler.py]: Please resolve pylint issues

Comment (SVilgelm) [inline tests/scheduler/test_scheduler.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.cli.v1.commands
cloudferry/cli/v1/commands.py:18: [E0401(import-error), ] Unable to import 'fabric.api'
cloudferry/cli/v1/commands.py:19: [E0401(import-error), ] Unable to import 'cliff'
************* Module tests.scheduler.test_scheduler
tests/scheduler/test_scheduler.py:16: [E0401(import-error), ] Unable to import 'mock'
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:21: [E0401(import-error), ] Unable to import 
'novaclient.v2'
cloudferry/lib/os/compute/nova_compute.py:22: [E0401(import-error), ] Unable to import 
'novaclient'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3887/

**Comment by SVilgelm** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/actions/compute/vm_restore.py [hunk 4]
```diff
@@ -34,19 +34,22 @@ def run(self, info, **kwargs):
...
-                instance_id=instance_id)
+        info_backup = kwargs.get('info_backup')
+        if info_backup:
+            for instance_id in src_compute.processing_instances:
+                instance = info_backup['instances'][instance_id]['instance']
+                LOG.debug("Status of '%s' (%s) is changed to original state "
+                          "%s on SRC", instance['name'], instance_id,
+                          instance['status'])
+                # reset status of vm
+                src_compute.change_status(
```
Comment (antonf) [inline cloudferry/actions/compute/vm_restore.py]: This loop will be interrupted if change status will raise some exception. You should catch them, but don't interrupt the loop

**Comment by antonf** [comment]
-1

File: cloudferry/actions/compute/vm_restore.py (hunk 4 above)
Comment (SVilgelm) [inline cloudferry/actions/compute/vm_restore.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/actions/compute/vm_restore.py [hunk 5]
```diff
@@ -34,19 +34,28 @@ def run(self, info, **kwargs):
...
+                instance = info_backup['instances'][instance_id]['instance']
+                LOG.debug("Status of '%s' (%s) is changed to original state "
+                          "%s on SRC", instance['name'], instance_id,
+                          instance['status'])
+                # reset status of vm
+                try:
+                    src_compute.change_status(
+                        instance['status'],
+                        instance_id=instance_id)
+                except Exception as e:  # pylint: disable=broad-except
```
Comment (antonf) [inline cloudferry/actions/compute/vm_restore.py]: Isn't it ok to catch only nova client exceptions here?

Comment (SVilgelm) [inline cloudferry/actions/compute/vm_restore.py]: i think `Nova`, `AbortMigration` and `TimeoutException` should be enough.

Comment (SVilgelm) [inline cloudferry/actions/compute/vm_restore.py]: We don't need `AbortMigrationError`

Comment (SVilgelm) [inline cloudferry/actions/compute/vm_restore.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.actions.compute.vm_restore
cloudferry/actions/compute/vm_restore.py:17: [E0401(import-error), ] Unable to import 
'novaclient'
************* Module cloudferry.cli.v1.commands
cloudferry/cli/v1/commands.py:18: [E0401(import-error), ] Unable to import 'fabric.api'
cloudferry/cli/v1/commands.py:19: [E0401(import-error), ] Unable to import 'cliff'
************* Module tests.scheduler.test_scheduler
tests/scheduler/test_scheduler.py:16: [E0401(import-error), ] Unable to import 'mock'
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:21: [E0401(import-error), ] Unable to import 
'novaclient.v2'
cloudferry/lib/os/compute/nova_compute.py:22: [E0401(import-error), ] Unable to import 
'novaclient'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
👍

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.scheduler.test_scheduler
tests/scheduler/test_scheduler.py:16: [E0401(import-error), ] Unable to import 'mock'

**Comment by SVilgelm** [comment]
retest

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**